apps/admin/frontend/src/screens/settings_screen.tsx

```diff
@@ -1,25 +1,42 @@
 import React, { useContext } from 'react';
+import { assert } from '@votingworks/basics';
 import {
   CurrentDateAndTime,
   Prose,
   RebootFromUsbButton,
   RebootToBiosButton,
   SetClockButton,
 } from '@votingworks/ui';
+import { isSystemAdministratorAuth } from '@votingworks/utils';
 
 import { AppContext } from '../contexts/app_context';
 import { NavigationScreen } from '../components/navigation_screen';
 import { FormatUsbButton } from '../components/format_usb_modal';
+import { updateSessionExpiry } from '../api';
 
 export function SettingsScreen(): JSX.Element {
-  const { logger, usbDrive } = useContext(AppContext);
+  const { auth, logger, usbDrive } = useContext(AppContext);
+  const updateSessionExpiryMutation = updateSessionExpiry.useMutation();
+
+  assert(isSystemAdministratorAuth(auth));
 
   return (
     <NavigationScreen title="Settings">
       <Prose maxWidth={false}>
         <h2>Current Date and Time</h2>
         <p>
-          <SetClockButton>
+          <SetClockButton
+            sessionExpiresAt={auth.sessionExpiresAt}
+            updateSessionExpiry={async (sessionExpiresAt: Date) => {
+              try {
+                await updateSessionExpiryMutation.mutateAsync({
+                  sessionExpiresAt,
+                });
+              } catch {
+                // Handled by default query client error handling
+              }
```

Comment:
I know this pattern is clunky, but it's currently the best option we have for passing promisified react-query mutations to reusable components while also preventing unhandled promise rejections.

Touched on in [this presentation slide](https://docs.google.com/presentation/d/1qcohhkeavCVjTZAAhGbbB7sbnvDJLlNov3OTLijfcvs/edit#slide=id.g1f8414d696e_0_361) from a while back